overlord/snapstate/handlers.go

```diff
@@ -1341,6 +1341,21 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		snapst.Sequence[len(snapst.Sequence)-1] = cand
 	}
 
+	// update snapst.RevertStatus map if needed
+	if snapsup.Revert {
+		switch snapsup.RevertStatus {
+		case NotBlocked:
+			if snapst.RevertStatus == nil {
+				snapst.RevertStatus = make(map[int]RevertStatus)
+			}
+			snapst.RevertStatus[snapst.Current.N] = NotBlocked
```

Comment:
don't we need to something in undoLinkSnap to undo this?

Reply:
Added.